File: implementation/scipamato/scipamato-common-entity/src/test/java/ch/difty/scipamato/common/entity/CodeClassIdTest.java
package ch.difty.scipamato.common.entity;

import static ch.difty.scipamato.common.entity.CodeClassId.*;
import static org.assertj.core.api.Assertions.*;

import org.junit.Test;

public class CodeClassIdTest {

    @Test
    public void values() {
        assertThat(CodeClassId.values()).containsExactly(CC1, CC2, CC3, CC4, CC5, CC6, CC7, CC8);
    }

    @Test
    public void fromId_whenPresent() {
        assertThat(CodeClassId.fromId(1).isPresent()).isTrue();
        assertThat(CodeClassId.fromId(1).get()).isEqualTo(CC1);
        assertThat(CodeClassId.fromId(2).get()).isEqualTo(CC2);
        assertThat(CodeClassId.fromId(3).get()).isEqualTo(CC3);
        assertThat(CodeClassId.fromId(4).get()).isEqualTo(CC4);
        assertThat(CodeClassId.fromId(5).get()).isEqualTo(CC5);
        assertThat(CodeClassId.fromId(6).get()).isEqualTo(CC6);
        assertThat(CodeClassId.fromId(7).get()).isEqualTo(CC7);
        assertThat(CodeClassId.fromId(8).get()).isEqualTo(CC8);
    }

    @Test
    public void fromId_whenNotPresent() {
        assertThat(CodeClassId.fromId(-1).isPresent()).isFalse();
    }

    @Test
    public void getId() {
        assertThat(CodeClassId.CC1.getId()).isEqualTo(1);
    }

}

File: common/common-entity/src/test/kotlin/ch/difty/scipamato/common/entity/CodeClassIdTest.kt
package ch.difty.scipamato.common.entity

import ch.difty.scipamato.common.entity.CodeClassId.CC1
import ch.difty.scipamato.common.entity.CodeClassId.CC2
import ch.difty.scipamato.common.entity.CodeClassId.CC3
import ch.difty.scipamato.common.entity.CodeClassId.CC4
import ch.difty.scipamato.common.entity.CodeClassId.CC5
import ch.difty.scipamato.common.entity.CodeClassId.CC6
import ch.difty.scipamato.common.entity.CodeClassId.CC7
import ch.difty.scipamato.common.entity.CodeClassId.CC8
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldBeFalse
import org.amshove.kluent.shouldBeTrue
import org.amshove.kluent.shouldContainSame
import org.junit.jupiter.api.Test

class CodeClassIdTest {

    @Test
    fun values() {
        CodeClassId.entries shouldContainSame listOf(CC1, CC2, CC3, CC4, CC5, CC6, CC7, CC8)
    }

    @Test
    fun fromId_whenPresent() {
        CodeClassId.fromId(1).isPresent.shouldBeTrue()
        CodeClassId.fromId(1).get() shouldBeEqualTo CC1
        CodeClassId.fromId(2).get() shouldBeEqualTo CC2
        CodeClassId.fromId(3).get() shouldBeEqualTo CC3
        CodeClassId.fromId(4).get() shouldBeEqualTo CC4
        CodeClassId.fromId(5).get() shouldBeEqualTo CC5
        CodeClassId.fromId(6).get() shouldBeEqualTo CC6
        CodeClassId.fromId(7).get() shouldBeEqualTo CC7
        CodeClassId.fromId(8).get() shouldBeEqualTo CC8
    }

    @Test
    fun fromId_whenNotPresent() {
        CodeClassId.fromId(-1).isPresent.shouldBeFalse()
    }

    @Test
    fun getId() {
        CC1.id shouldBeEqualTo 1
    }
}
